test_semver.py

```diff
@@ -655,3 +656,52 @@ def test_should_process_raise_error(capsys):
     assert rc != 0
     captured = capsys.readouterr()
     assert captured.err.startswith("ERROR")
+
+
+@pytest.mark.parametrize("version,parts,expected", [
+    ("3.4.5", dict(major=2), '2.4.5'),
+    ("3.4.5", dict(major="2"), '2.4.5'),
+    ("3.4.5", dict(major=2, minor=5), '2.5.5'),
+    ("3.4.5", dict(minor=2), '3.2.5'),
+    ("3.4.5", dict(major=2, minor=5, patch=10), '2.5.10'),
+    ("3.4.5", dict(major=2, minor=5, patch=10, prerelease="rc1"),
+     '2.5.10-rc1'),
+    ("3.4.5", dict(major=2, minor=5, patch=10, prerelease="rc1", build="b1"),
+     '2.5.10-rc1+b1'),
+    ("3.4.5-alpha.1.2", dict(major=2), '2.4.5-alpha.1.2'),
+    ("3.4.5-alpha.1.2", dict(build="x1"), '3.4.5-alpha.1.2+x1'),
+    ("3.4.5+build1", dict(major=2), '2.4.5+build1'),
+])
+def test_replace_method_replaces_requested_parts(version, parts, expected):
+    assert replace(version, **parts) == expected
+
+
+def test_replace_raises_TypeError_for_invalid_keyword_arg():
+    with pytest.raises(TypeError):
```

Comment:
You can use the following to avoid having a 2nd test case for the same:

```suggestion
    with pytest.raises(TypeError, match=r"replace\(\).*unknown.*"):
```

See http://doc.pytest.org/en/latest/assert.html#assertions-about-expected-exceptions

Reply:
Brilliant! :smile: I searched for a better way, now as you mention it. :+1: It's now implemented in commit 5049b83